packages/z3-static/z3-static.4.8.13/opam

```diff
@@ -0,0 +1,38 @@
+opam-version: "2.0"
+maintainer: "simon@aestheticintegration.com"
+authors: "MSR"
+homepage: "https://github.com/Z3prover/z3"
+bug-reports: "https://github.com/Z3prover/z3/issues"
+license: "MIT"
+dev-repo: "git+https://github.com/Z3prover/z3.git"
+build: [
+  [ "python2.7" "scripts/mk_make.py" "--ml" "--staticlib" ]
+  [ make "-C" "build" "-j" jobs ]
+]
+
+install: [
+  [ "sh" "-c" "ocamlfind install z3 build/api/ml/* build/libz3-static.a"]
+  [ "cp" "build/z3" "%{bin}%/z3-static"]
```

Comment:
Now, the final question with this PR is whether the name of the binary for statically-linked installation of z3 should be called `z3-static` or just `z3`? Note that the package is still called z3 here.

cc: @arbipher @c-cube

Reply:
Should there be a mutual conflict between the packages?

Reply:
oh, @k4rtik you are right. they should be `z3` and `z3-static`. This was also discussed in that same [z3 issue](https://github.com/Z3Prover/z3/issues/5578).

@mseri these two packages are consistent and they can be both installed on ubuntu. `z3` and `z3-static` relies on no system-level z3, both are built from the source and there should be no name conflict based on my last fix.

Reply:
I guess I am confused about whether the package name in `opam list` should be `z3` or `z3-static` after doing the static install. In my testing, it came out to be `z3` but that may have been because of the way I am doing testing (manually downloading the tarball, copying the opam file in the extracted directory whose name is `z3.4.8.13` and then issuing `opam install ./opam`). But if this is not what we want, then do the users need to use the statically linked z3 using a different command:

```
ocamlfind ocamlc -o ml_example.byte -thread -package z3-static -linkpkg ml_example.ml
```
instead of the [usual](https://github.com/Z3Prover/z3/tree/master/examples/ml):
```
ocamlfind ocamlc -o ml_example.byte -thread -package z3 -linkpkg ml_example.ml
```
If this is the case, we need to document it properly.

Can someone else verify the static install on Linux and report back?

Reply:
There will be two distinct packages `z3` and `z3-static`.

`opam list` should have `z3` if you install `z3`, `z3-static` if you install `z3-static`. I guess your manual installation may leave some files in the opam site path.

This is my testing [file](https://github.com/arbipher/ocaml-build-examples/blob/main/Makefile), which tests the combination of using `{z3,z3-static}` * `{bytecode, native, bytecode-with-custom}`. Every use of the z3 static should use `z3-static`. `z3` and `z3-static` shares nothing.

btw, after changing line 14 to 
`[ "sh" "-c" "ocamlfind install z3-static build/api/ml/* build/libz3-static.a"]`,
it could pass all my tests with ocaml 4.13.1 on my ubuntu.